src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,178 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node first;
+    private Node last;
+
+    private int size;
+
+    private static class Node<T> {
+        private T element;
+        private Node next;
+        private Node previous;
+
+        public Node(T element) {
+            this.element = element;
+        }
+    }
+
     @Override
-    public void add(T value) {
+    public void add(T element) {
+        Node newNode = new Node(element);
+        if (first == null) {
+            newNode.next = null;
+            newNode.previous = null;
+            first = newNode;
+            last = newNode;
+        } else {
+            last.next = newNode;
+            newNode.previous = last;
+            last = newNode;
+        }
+        size++;
     }
 
     @Override
-    public void add(T value, int index) {
+    public void add(T element, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException();
+        }
+        Node<T> newNode = new Node(element);
+        if (index == size) {
+            add(element);
+        } else if (index == 0) {
+            Node<T> oldNode = first;
+            oldNode.previous = newNode;
+            newNode.next = oldNode;
+            first = newNode;
+
+            size++;
+        } else {
+            Node<T> oldNode = first;
+            for (int i = 0; i < index; i++) {
+                oldNode = oldNode.next;
+            }
+            Node<T> oldPrevious = oldNode.previous;
```

Comment:
lets have a separate method ```getNode(index)``` it will be useful in here

also https://mate-academy.github.io/jv-program-common-mistakes/java-core/collections/linked-list.html#dont-iterate-your-nodes-from-the-head-to-tail-or-vice-versa